src/main/windows/editor.js

```diff
@@ -114,94 +149,294 @@ class EditorWindow extends BaseWindow {
 
     // The window is now destroyed.
     win.on('closed', () => {
+      this.lifecycle = WindowLifecycle.QUITTING
       this.emit('window-closed')
 
       // Free window reference
       win = null
     })
 
-    win.loadURL(this._buildUrlWithSettings(this.id, env, preferences))
+    this.lifecycle = WindowLifecycle.LOADING
+    win.loadURL(this._buildUrlString(this.id, env, preferences))
     win.setSheetOffset(TITLE_BAR_HEIGHT)
 
+    mainWindowState.manage(win)
+
+    setTimeout(() => {
```

Comment:
Why need to use setTimeout?

Reply:
I don't really know if we need this but the idea behind this is to process other stuff that is needed to load the window and after application start first and then load files. So we queue it after the current operation is completed (node event loop).

Reply:
Pls add some comments.